### PR TITLE
Remove [Obsolete] attribute for ConsulClient HttpClient constructor

### LINKED ISSUE #24: Support for injecting HttpClient in a non-obsolete constructor
As of now ConsulClient allows for injecting an HttpClient using an [Obsolete] constructor ([here](https://github.com/G-Research/consuldotnet/blob/master/Consul/Client.cs#L446)). This project should remove this [Obsolete] attribute until an alternative way to inject HttpClient is provided as a migration path (my suggestion is sticking with the current constructor or just a Func\<HttpClient\>). As of now the non-obsolete constructors carry the issue that they internally manage an HttpClient, which means that you either dispose ConsulClient and risk socket exhaustion, or you use a singleton ConsulClient and risk dealing with a stale DNS (see [here](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests)). My personal opinion is that ConsulClient should not be managing HttpClient lifetimes at all since they hide serious issues from the library user.

### PR DESCRIPTION
**Pull request by Salgat**
This is to allow continued support for providing an external HttpClient to the ConsulClient. Closes #24

### CODE REVIEW

**Review by lorinck — DISMISSED** [review]
LGTM as discussed under https://github.com/G-Research/consuldotnet/issues/24.
Thank you for raising the PR @Salgat

**Comment by jgiannuzzi** [comment]
@Salgat @lorinck @maurosbarigia I wonder whether we shouldn't also un-obsolete the `ConsulClient(ConsulClientConfiguration config)` constructor?

I understand that @Salgat wants to be able to pass an external `HttpClient` to the constructor, but leaving the other constructor obsolete does not sound consistent. What do you think?

**Comment by Salgat** [comment]
If uniformity is desired it may be more appropriate to leave both obsolete and add a "ConsulClient(Action<ConsulClientConfiguration> configOverride, HttpClient httpClient)" constructor, but I'm good with either way since this is an opinionated decision.

**Comment by Salgat** [comment]
Would it be possible to merge this minor change in for now? This can always be updated later when/if a decision is made on what constructors are ultimately decided on. The goal of this PR is more intent than anything else (since it does not change the behavior of anything, only prevents removing this constructor until a better alternative is present).

File: Consul/Client.cs [hunk 1]
```diff
@@ -443,8 +443,6 @@ public ConsulClient(ConsulClientConfiguration config)
         /// </summary>
         /// <param name="config">A Consul client configuration</param>
         /// <param name="client">A custom HttpClient</param>
```
Comment (marcin-krystianc) [inline Consul/Client.cs:445]: It would be nice to add a comment, that it is responsibility of the user to handle the lifecycle of `httpclient`.

**Review by marcin-krystianc — COMMENTED** [review]
@Salgat apologies for the late response. Before we merge this, can you bump the [version number](https://github.com/G-Research/consuldotnet/blob/master/Consul/Consul.csproj#L4) and add a new entry to the [changelog](https://github.com/G-Research/consuldotnet/blob/master/CHANGELOG.md) ?

**Comment by jgiannuzzi** [comment]
> @Salgat apologies for the late response. Before we merge this, can you bump the [version number](https://github.com/G-Research/consuldotnet/blob/master/Consul/Consul.csproj#L4) and add a new entry to the [changelog](https://github.com/G-Research/consuldotnet/blob/master/CHANGELOG.md) ?

@Salgat In the meantime we changed this guidance, could you update/rebase your branch and simply add an entry to the [changelog](https://github.com/G-Research/consuldotnet/blob/master/CHANGELOG.md)?

**Comment by Salgat** [comment]
Sorry for the delay! I'll see about updating this later today or tomorrow.

**Comment by Salgat** [comment]
Alright should be all good now. Let me know if any other changes are needed.

File: Consul/Client.cs (hunk 1 above)
Comment (Salgat) [reply Consul/Client.cs]: Great suggestion @marcin-krystianc I updated the documentation for this parameter.

**Review by lorinck — APPROVED** [review]

**Comment by lorinck** [comment]
Thanks for raising @Salgat

**Comment by jgiannuzzi** [comment]
Thanks a lot for your contribution to Consul.NET, @Salgat!

**Pull request merged**